src/tools/rbd_mirror/group_replayer/Replayer.cc

```diff
@@ -1259,18 +1259,11 @@ void Replayer<I>::handle_create_regular_snapshot(
     dout(10) << "cleaning snapshot as failed group_snap_set previously with : "
              << group_snap_id << ", this will be attempted to sync later"
              << dendl;
-    r = librbd::cls_client::group_snap_remove(&m_local_io_ctx,
+    int ret = librbd::cls_client::group_snap_remove(&m_local_io_ctx,
```

Comment:
Why does `group_snap_remove` need to be called if the call to create the snapshot failed?

Reply:
> Why does group_snap_remove need to be called if the call to create the snapshot failed?

I'm not completely certain, Won't there be any partial data written in any case of failure ?

Reply:
Suggestion on commit message:

rbd-mirror: don't erase local snap vector elements
1. Fixes a crash in handle_create_regular_snapshot caused by accessing an invalid iterator.
2. Fixes return variables being overwritten.

Reply:
I thought we are doing this push_back in many other places, if this is the only place, then probably it should be easy to address. I will take a relook and get back.

Reply:
> For the first call to group_snap_set() (i.e. one that actually creates a GROUP_SNAPSHOT_STATE_INCOMPLETE entry for the snapshot), "any partial data written" is impossible. At the OSD level it's a proper transaction -- it's either applied or not. I don't think group_snap_remove() needs to be called in such cases.

I see it now. Thanks.